### PR TITLE
Sf6 compatibility

### PR DESCRIPTION
**Pull request by tobias-93**
I didn't want to hijack pull request #1237, but I have fixed the tests for Symfony 5.4 and removed the PHP 8.1 deprecations.
Symfony 6.0 cannot be tested because `knplabs/knp-gaufrette-bundle` is not yet SF6 compatible (this was not found in the previous tests, since that didn't properly fix the Symfony version leading to some components still using the 5.4 version).
If anyone using SF6 already could test this I would be very grateful.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -33,52 +33,69 @@ jobs:
...
-                      symfony: 5.2.*
-                      validate_docs: false
+                    - php: '8.0'
+                      symfony: '5.4.*'
+                      dependencies: highest
                       with_imagine: false
                       with_oneupbundle: true
-        name: PHP ${{ matrix.php }} tests (${{ matrix.description }})
+
+# Cannot be enabled yet because knplabs/knp-gaufrette-bundle doesn't support SF6 yet
```
Comment (garak) [inline .github/workflows/build.yaml:58]: Indeed, knplabs/knp-gaufrette-bundle supports sf6 in 0.7.x-dev

**Review by garak — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (tobias-93) [reply .github/workflows/build.yaml]: Using that now, thanks

**Review by garak — APPROVED** [review]

**Comment by sxsws** [comment]
still can't install with symfony 6.0.1 and php 8.1?

**Comment by garak** [comment]
For now, you need to use our branch alias

**Comment by sxsws** [comment]
which branch alias?

File: composer.json [hunk 2]
```diff
@@ -18,14 +18,14 @@
...
         "symfony/event-dispatcher-contracts": "^1.1 || ^2.0",
-        "symfony/http-foundation": "^4.4 || ^5.0",
-        "symfony/http-kernel": "^4.4 || ^5.0",
-        "symfony/mime": "^4.4 || ^5.0",
-        "symfony/property-access": "^4.4 || ^5.0",
+        "symfony/http-foundation": "^4.4 || ^5.0 || ^6.0",
+        "symfony/http-kernel": "^4.4 || ^5.0 || ^6.0",
+        "symfony/mime": "^4.4 || ^5.0 || ^6.0",
+        "symfony/property-access": "^4.4 || ^5.0 || ^6.0",
         "symfony/string": "^5.0"
```
Comment (garak) [inline composer.json:29]: this was forgotten...

File: composer.json [hunk 3]
```diff
@@ -34,27 +34,27 @@
...
-        "symfony/var-dumper": "^4.4 || ^5.0",
-        "symfony/yaml": "^4.4 || ^5.0"
+        "symfony/asset": "^4.4 || ^5.0 || ^6.0",
+        "symfony/browser-kit": "^4.4 || ^5.0 || ^6.0",
+        "symfony/css-selector": "^4.4 || ^5.0 || ^6.0",
+        "symfony/doctrine-bridge": "^4.4 || ^5.0 || ^6.0",
+        "symfony/dom-crawler": "^4.4 || ^5.0 || ^6.0",
+        "symfony/form": "^4.4 || ^5.0 || ^6.0",
+        "symfony/framework-bundle": "^4.4 || ^5.0 || ^6.0",
+        "symfony/phpunit-bridge": "^5.4",
```
Comment (garak) [inline composer.json:50]: this one was forgotten, too

File: composer.json (hunk 2 above)
Comment (sxsws) [reply composer.json:29]: and contracts too

Comment (tobias-93) [reply composer.json:29]: See #1251, thanks for the feedback

File: composer.json (hunk 3 above)
Comment (tobias-93) [reply composer.json:50]: See #1251

**Pull request merged**